src/socket.io/categories.js

```diff
@@ -203,10 +203,10 @@ function ignoreOrWatch(fn, socket, cid, callback) {
 
 			async.each(cids, function (cid, next) {
 				fn(socket.uid, cid, next);
-			}, next);
+			}, next(null, cids));
 		},
-		function (next) {
-			topics.pushUnreadCount(socket.uid, next);
+		function (cids, next) {
+			topics.pushUnreadCount(socket.uid, next(null, cids));
```

Comment:
same as above, this passes undefined to `pushUnreadCount` callback, just move the `cids` declaration out of async waterfall.

Reply:
I did that to send the changed cids to the page script to update the watched/ignored buttons, this way I don't need to compute or keep track of who is child of who in the page as that's already done in this function. I'll take a look in the weekend but if you can give an idea or hint it would be very welcome.

Reply:
You can move `var cids = [parseInt(cid, 10)];` to line 177 and then pass it to the client side. See below.

```
function ignoreOrWatch(fn, socket, cid, callback) {
        var cids = [parseInt(cid, 10)];
	async.waterfall([
		function (next) {
			db.getSortedSetRange('categories:cid', 0, -1, next);
		},
		function (cids, next) {
			categories.getCategoriesFields(cids, ['cid', 'parentCid'], next);
		},
		function (categoryData, next) {
			categoryData.forEach(function (c) {
				c.cid = parseInt(c.cid, 10);
				c.parentCid = parseInt(c.parentCid, 10);
			});

			// filter to subcategories of cid
			var cat;
			do {
				cat = categoryData.find(function (c) {
					return cids.indexOf(c.cid) === -1 && cids.indexOf(c.parentCid) !== -1;
				});
				if (cat) {
					cids.push(cat.cid);
				}
			} while (cat);

			async.each(cids, function (cid, next) {
				fn(socket.uid, cid, next);
			}, next);
		},
		function (next) {
			topics.pushUnreadCount(socket.uid, next(null, cids));
		},
		function (next) {
			next(null, cids);
		}
	], callback);
}

```